lib/rdoc/yard.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+##
+# YARD compatibility module for RDoc
+#
+# This module provides support for parsing YARD tags in RDoc comments.
+# Currently supports:
+# - @yield [params] - Describes block parameters
+# - @private - Marks method/class as private
+# - @api private - Alternative private marking
+
+class RDoc::YARD
+
+  ##
+  # Processes YARD tags in the given comment text for a code object
+  #
+  # Returns modified comment text with YARD tags processed
+  def self.process(comment, code_object)
+    new(comment, code_object).process
+  end
+
+  attr_reader :comment, :code_object, :text
+
+  def initialize(comment, code_object)
+    @comment = comment
+    @code_object = code_object
+  end
+
+  def process
+    return comment unless comment && code_object
+
+    # Only process RDoc::Comment objects to avoid modifying plain strings
+    # that aren't actually documentation comments
+    return comment unless comment.is_a?(RDoc::Comment)
+
+    # Skip processing if comment is in TomDoc format
+    return comment if comment.format == 'tomdoc'
```

Comment:
Can we use `comment.tomdoc?`

BTW, is this a responsibility of this class? It seems that this is a responsibility of callers.